### PR TITLE
Add attached domain example

### PR DESCRIPTION
**Pull request by VincentCasse**
One more example to help customer to migrate from SoAPI to APIv6

Signed-off-by: Vincent Casse <vincent.casse@corp.ovh.com>

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+PHP wrapper examples
+--------------------
+
+In this part, you can found real use case of the OVH php wrapper
```
Comment (yadutaf) [inline examples/README.md]: - s/found/find/g
- s/use case of/use cases for/

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+How to attach domains to web hosting offer using php wrapper?
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: - to *a* web hosting
- using *the* php

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+-------------------------------------------------------------
+
+This documentation will help you to create, edit and delete attached domains to your webhosting offer. This documentation is the equivalent of [MultiDomain SoAPI](http://www.ovh.com/soapi/en/?method=multiDomainAdd) and [SubDomain SoAPI](http://www.ovh.com/soapi/en/?method=subDomainAdd)
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: domains attached

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Compatibility note
+
+MultiDomains and SubDomains options are merged into one option called **Attached Domains**. You can set up subdomains and other domains to your web hosting with this new option.
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: You can manage sub domains and muti domains using this option (?)

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+- Having PHP 5.2+
+- Having a domain at OVH
+- Having an hosting account
+
+## Download PHP wrapper
+
+- Download the latest release **with dependencies** on github: https://github.com/ovh/php-ovh/releases
+
+```bash
+# When this article is written, latest version is 2.0.0
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: - *was* written
- latest version *was*

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+wget https://github.com/ovh/php-ovh/releases/download/v2.0.0/php-ovh-2.0.0-with-dependencies.tar.gz
+```
+
+- Extract it into a folder
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: "Extract it" (no "into ..."?)

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+```bash
+tar xzvf php-ovh-2.0.0-with-dependencies.tar.gz 
+```
+
+## Create a new token
+You can create a new token using [this url](https://api.ovh.com/createToken/?GET=/hosting/web/my_domain/attachedDomain&POST=/hosting/web/my_domain/attachedDomain&GET=/hosting/web/my_domain/attachedDomain/*&PUT=/hosting/web/my_domain/attachedDomain/*&DELETE=/hosting/web/my_domain/attachedDomain/*). Keep application key, application secret and consumer key to complete the script.
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: Kepp the* application ...

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+Be warn, this token is only validated for this script and for hosting called **my_domain**. Please replace **my_domain** by your web hosting reference!
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: - "warning" or "be warned"
- "is only valid* for"

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+If you need a more generic token, you had to change right field.
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: s/you had to change right field/make sure to set the rights field to your needs/

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+## Attach a domain to your web hosting
+
+When you call the API to attach a domain to your web hosting, the api call returns a **task**. The task is the current state of an operation to attach this domain to your hosting. The [example script](createAttachedDomain.php) explains how to attach a domain an wait the end of the operation.
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: - "The task holds* the current state" (?)
- "attach a domain and* wait "

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+If this script works, you will show something like:
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: you should see someting like

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'doing'
+Domain attached to the web hosting
+```
+
+## List all your attached domains
+
+The [example script](listAttachedDomain.php) explains how to show all your domains attached to an web hosting
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: attached to a* web hosting

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+For instance, using the example values in this script, the answer could be look like:
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: you should (or "will", as you prefer) see something like

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    [firewall] => none
+    [path] => otherFolder
+)
+```
+
+## Detach a domain from your web hosting
+
+The [example script](deleteAttachedDomain.php) explains how to detach a domain to your web hosting.
+
+If this script works, you will show something like:
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: you should see someting like

File: examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'todo'
+Status of task #42 is 'doing'
+Domain detached from the web hosting
+```
+
+## What's more?
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/api_attach_domain_to_web_hosting.md]: what's next ? or "getting further" ?

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+// Informations about your application
+$applicationKey = "your_app_key";
+$applicationSecret = "your_app_secret";
+$consumer_key = "your_consumer_key";
+
+// Information about API and rights asked
+$endpoint = 'ovh-eu';
+
+// Information about your web hosting and the attached domain (compulsory)
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: Informations*

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+$domain = 'mydomain.ovh'; // Web hosting id (often domain order with it)
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: ordered*

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+		'domain'	=> $domainToAttach,
+		'path'		=> $path,
+		'cdn'		=> $cdn,
+		'firewall'	=> $firewall,
+		'ownLog'	=> $ownLog,
+	));
+
+	echo "Task #" . $task['id'] . " is created" . PHP_EOL;
+
+	// we check each 5 seconds if task are ended
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: we check every* 5 seconds if the* task is* done*

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+	// When the task disappear, the task is done
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: s/disappear/disappears/

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+	while ( 1 ) {
+		try {
+			$wait = $conn->get('/hosting/web/' . $domain . '/tasks/' . $task['id']);
+			
+			if ( strcmp( $wait['status'], 'error' ) === 0 ) {
+				// The task is in error state. Please check your parameters, retry or contact support.
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: I can afford telling for every single lines, but your comments are really helpful. That's a really good work. Thanks!

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+				echo "An error is occured during the task" . PHP_EOL;
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: s/is/has/

File: examples/hosting-attachedDomain/createAttachedDomain.php [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+				break;
+			} elseif ( strcmp( $wait['status'], 'cancelled' ) === 0 ) {
+				// The task is in cancelled state. Please check your parameters, retry or contact support.
+				echo "Task is cancelled during the task" . PHP_EOL;
```
Comment (yadutaf) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: s/is/has been/

File: examples/hosting-attachedDomain/createAttachedDomain.php (hunk 20 above)
Comment (VincentCasse) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: :)

**Comment by yadutaf** [comment]
LGTM

File: examples/hosting-attachedDomain/createAttachedDomain.php (hunk 20 above)
Comment (hugsbrugs) [inline examples/hosting-attachedDomain/createAttachedDomain.php]: Whaou ! Need to catch an error to know task has been successfully done : what a beautiful design of an API. If an error occurs : retry or contact support ... would'nt an error message be better ? Your script will never work if you have not set A record and TXT record on domain you want to attached to your hosting ... but had to find that on my own !

**Pull request merged**